src/modules/flow/persistence/persistence.c

```diff
@@ -320,23 +321,31 @@ persist_open(struct sol_flow_node *node,
     SOL_NULL_CHECK(mdata->name, -ENOMEM);
 
     /* a zero packet_data_size means dynamic size content */
+    r = storage_read(mdata, &buf);
     if (mdata->packet_data_size) {
-        struct sol_buffer buf = SOL_BUFFER_INIT_FLAGS(mdata->value_ptr,
-            mdata->packet_data_size,
-            SOL_BUFFER_FLAGS_MEMORY_NOT_OWNED | SOL_BUFFER_FLAGS_NO_NUL_BYTE);
-
-        r = storage_read(mdata, &buf);
+        if (r >= 0) {
+            /* entry's total size may be bigger that actual
+             * packet_data_size (think bit fields). the useful data
+             * with be the leading bytes, on all cases */
```

Comment:
I didn't get this... it shouldn't be the other way round: entry size smaller than packet size?